### ISSUE TITLE
Mouse Capture Not Working M1 Mac in Full Screen

### ISSUE THREAD

**Issue by nyteshade** [BlitterStudio/amiberry#995]
**Describe the bug**
Mouse capture is not working in full screen mode (and buggy in windowed mode) 

**To Reproduce**

1. Run a configuration of your choosing
2. Select full screen widget either by menu or full screen button on window
3. Move mouse (usually up and down trigger the fastest) around a bit
4. Notice that the Dock or sometimes Mac menu show up unexpectedly
5. Can accidentally trigger quit of Amiberry when trying to click Amiga menus

**Expected behavior**
Movement of the mouse when the emulator is active should never interact with macOS widgets, widows or buttons, maybe excepting macOS modal windows that somehow get triggered while using the emulator.

**Screenshots**
None

**Desktop (please complete the following information):**

- OS: macOS Monterey 12.4, Apple M1 Pro, 16GB Memory

- Browser: n/a

- Version: 5.3 from GitHub as of 7/July/2022

**Additional context**

Love Amiberry? Please consider supporting our collective:
👉  https://opencollective.com/amiberry/donate

**Comment by midwan** [comment]
@solskogen can you reproduce this?

**Comment by solskogen** [comment]
I cannot, but (and this might me related) if I restart amiberry after emulation has started the mouse doesn't show in the gui. I suspect this is an SDL issue.

**Comment by corasaniti** [comment]
Has anyone solved the problem? I have the same issue with Amiberry 5.3 on Dietpi 8.8. 
The mouse is not captured in full screen mode. 
I downgraded to version 5.2 and everything started working again.
Thanks

**Comment by solskogen** [comment]
what version of SDL do you have installed?

**Comment by corasaniti** [comment]
@solskogen 
How can I see it? Thank you

**Comment by solskogen** [comment]
I'm not sure how the package manager on DietPi works.

**Comment by MichaIng** [comment]
On DietPi, we found this being an issue with libSDL v2.24.0. Downgrading it to v2.0.22 make it functional again. I'll do some more tests to see what I can get as debug info.

New packages for all platforms and distro versions are currently compiling, so @corasaniti I'll give you a ping when you can upgrade it (upgrade the package = downgrade the embedded libSDL 😉). Which SBC (in case of RPi, which architecture) and which Debian version is it?

**Comment by corasaniti** [comment]
@MichaIng 
I have these issue  with Raspberry Pi4, arch 32bit and Dietpi 8.8.2 Debian bullseye based.
I await your news toupdate to the Amiberry 5.3 version in the meantime I continue to use the 5.2
Thanks in advance

**Comment by midwan** [comment]
@corasaniti 
Interesting, are you saying that this behavior changed from 5.2 to 5.3? I don't think there was anything related between those versions, but if that's the case I'll have to take a closer look...

**Comment by MichaIng** [comment]
I'm sure it is the SDL version only. We never shipped Amiberry v5.2 in combination with SDL 2.24.0.

@corasaniti 
Please try:
```sh
cd /tmp
curl -O 'https://dietpi.com/downloads/binaries/bullseye/testing/amiberry_rpi4.deb'
dpkg -i amiberry_rpi4.deb
rm amiberry_rpi4.deb
```

**Comment by solskogen** [comment]
2.24.0 has mouse issues in emulation with KMSDRM.

**Comment by corasaniti** [comment]
@MichaIng @solskogen 
From command line How can I see the version of SDL that I have installed on my Dietpi system? 
Can I upgrade Amiberry to ver 5.3 and downgrade ONLY LibSDL to version 2.0.22 which seems to be the cause of the issue?
Thank you

**Comment by midwan** [comment]
> @MichaIng @solskogen From command line How can I see the version of SDL that I have installed on my Dietpi system? Can I upgrade Amiberry to ver 5.3 and downgrade ONLY LibSDL to version 2.0.22 which seems to be the cause of the issue? Thank you

Amiberry will show you which SDL2 version it was compiled and linked with, in the GUI -> About window.
You can run Amiberry with any version of SDL2, so if it works with an earlier one, perhaps that's a fair work-around for now.

I'll need to investigate the issue with KMSDRM and report it to the SDL team if I can recreate it.

**Comment by MichaIng** [comment]
@corasaniti 
The package I linked above is the combination of Amiberry 5.3 with SDL 2.0.22.

We compile and pack both into a single package, so the contained Amiberry binary and SDL so's always match and (usually) cannot be exchanged individually (without recompiling the binary).

**Comment by midwan** [comment]
I've reported the SDL2 2.24.0 + KMSDRM = no relative mouse motion issue upstream (https://github.com/libsdl-org/SDL/issues/6223)

**Comment by midwan** [comment]
@nyteshade I'm not sure if the fix above also fixes your issue on MacOS - please test and report back if possible?
If it's still an issue, we can re-open this case.

The KMSDRM related bug on RPI is fixed with the commit above.

**Comment by midwan** [comment]
@MichaIng 
This means that SDL2 2.24.0 can now be used again, with the latest commit. The fix will also be part of the 5.4 binary release.

**Comment by MichaIng** [comment]
Many thanks 👍. I'll wait for Amiberry v5.4 release to rebuild our packages with SDL2 v2.24.0 then.
